src/ecdsa/ecdsa.py

```diff
@@ -69,6 +68,26 @@ def __init__(self, r, s):
     self.r = r
     self.s = s
 
+  def recover_public_keys(self, hash, curve, g, n):
+    r = self.r
+    s = self.s
+    e = hash
+    x = r
+
+    # Compute the curve point with x as x-coordinate
+    alpha = ((x * x * x) + (curve.a() * x) + curve.b()) % curve.p()
```

Comment:
wouldn't it be more efficient to either use `pow(x, 3, curve.p())` (modulo exponentiation) or `(x * (x * x + curve.a()) + curve.b()) % curve.p()` (reduce number of multiplications)?